sycl/source/detail/scheduler/graph_builder.cpp

```diff
@@ -269,220 +271,114 @@ Command *Scheduler::GraphBuilder::addCopyBack(Requirement *Req) {
-            ^                               ^              acc creation
-            |                               |
-      +-------------+                +-------------+
-      |Write/Unmap 1|                |Write/Unmap 2|  <== This is blocked by
-      +-------------+                +-------------+      user event. Can be
-            ^                               ^             completed after host
-            +---------------+---------------+             accdesctruction
-                      +-------------+
-                      |  2 Kernel   |
-                      +-------------+
-
-    And the following content in OpenCL command queue:
-
-        +-------------------------------------------------------------------+
-    Q1: | 1K | Read/Map 1 | Write/Unmap 1 | Read/Map 2 | Write/Unmap 2 | 2K |
-        +-------------------------------------------------------------------+
-                                      ^                       ^
-                                      |                       |
-       This is blocked by user event -+-----------------------+
-
-    In the situation above there is "Write/Unmap 1" command already in
-    command queue which is blocked by user event and cannot be executed
-    and "Read/Map 2" command which is enqueued after "Write/Unmap 1" but
-    we should wait for the completion of this command before exiting
-    construction of the second host accessor.
-
-    Such cases is not supported in some OpenCL implementations. They
-    assume that the commands the are submitted before one user waits on
-    eventually completes.
-
-    This patch workarounds problem by using separate(exclusive) queues for
-    such tasks while still using one(common) queue for all other tasks.
-
-    So, for the second example SYCL RT creates 3 OpenCL queues, where
-    second and third queues are used for "Write/Unmap 1" and "Write/Unmap 2"
-    command respectively:
-
-        +-----------------------------------------------+
-    Q1: | 1 Kernel | Read/Map 1 | Read/Map 2 | 2 Kernel |
-        +-----------------------------------------------+
-
-        +---------------+
-    Q2: | Write/Unmap 1 |<----+
-        +---------------+     |
-                              |-----This is blocked by user event
-        +---------------+     |
-    Q3: | Write/Unmap 2 |<----+
-        +---------------+
-    */
-
-    std::unique_ptr<UnMapMemObject> UnMapCmdUniquePtr(new UnMapMemObject(
-        *SrcReq, SrcAllocaCmd, Req, SrcQueue, /*UseExclusiveQueue*/ true));
-
-    if (!MapCmdUniquePtr || !UnMapCmdUniquePtr)
-      throw runtime_error("Out of host memory");
-
-    MapMemObject *MapCmd = MapCmdUniquePtr.release();
+  OpType Type = OpType::COPY;
+
+  if (SrcQueue->is_host())
+    Type = UPDATE;
+  else if (SrcQueue->get_device().is_cpu() &&
+           Req->MSYCLMemObj->getType() ==
+               detail::SYCLMemObjI::MemObjType::BUFFER && false)
```

Comment:
As I can see this will always be `false`. Is it intentional?

Reply:
It is for testing. false will be removed from this line.